src/runtimes/node/index.js

```diff
@@ -4,20 +4,34 @@ const cpFile = require('cp-file')
 
 const { JS_BUNDLER_ESBUILD, JS_BUNDLER_ESBUILD_ZISI, JS_BUNDLER_ZISI, RUNTIME_JS } = require('../../utils/consts')
 
+const { detectEsModule } = require('./detect_es_module')
 const { findFunctionsInPaths } = require('./finder')
 const { getSrcFiles } = require('./src_files')
 const { zipEsbuild } = require('./zip_esbuild')
 const { zipZisi } = require('./zip_zisi')
 
 // We use ZISI as the default bundler, except for certain extensions, for which
 // esbuild is the only option.
-const getDefaultBundler = ({ extension }) =>
-  ['.mjs', '.ts'].includes(extension) ? JS_BUNDLER_ESBUILD : JS_BUNDLER_ZISI
+const getDefaultBundler = async ({ extension, mainFile, featureFlags }) => {
```

Comment:
[dust] If we add a default value to `featureFlags` we can probably lose the `&&` in line 20. What do you think?

```suggestion
const getDefaultBundler = async ({ extension, mainFile, featureFlags = {} }) => {
```